buildSrc/src/main/groovy/io.deephaven.java-toolchain-conventions.gradle

```diff
@@ -123,30 +123,54 @@ def devJvmArgs = [
 //  '-XX:+PrintCompilation',          // this optional line shows jit operations as they happen
 ]
 
+def createVmOptions = tasks.register('createVmOptions') {
+  def vmOptionsFile = project.layout.buildDirectory.file('dh-default.vmoptions')
+  it.outputs.file(vmOptionsFile)
+  doFirst {
+    // These are supposed to be generally applicable and recommended JVM options, but they aren't hard requirements.
+    // Overly specific options do *not* belong here. For example, we should _not_ be setting something like `-Xmx4g` here.
+    // If you are tempted to try and put system properties here (`-Dkey=value`), think again; there should be a more
+    // appropriate place to set those.
+    //
+    // From the perspective of our application distribution, these options will be used as defaults for JAVA_OPTS
+    // (if the user already has JAVA_OPTS set, these VM options will _not_ apply).
+    vmOptionsFile.get().asFile.text = "-XX:+UseG1GC\n-XX:MaxGCPauseMillis=100\n-XX:+UseStringDeduplication\n"
```

Comment:
isn't there a groovy-ism that can make this easier to read slash harder to break the `\n`s as we go? ideally we give it an array of individual options, it puts each on a new line, since that appears to be the expected format (rather than just join on whitespace)

Reply:
Done